packages/database/src/TupaiaDatabase.js

```diff
@@ -581,7 +581,10 @@ function addWhereClause(connection, baseQuery, where) {
     if (!VALID_COMPARISON_TYPES.includes(comparisonType)) {
       throw new Error(`Cannot compare using ${comparisonType}`);
     }
-    const columnSelector = castAs ? connection.raw(`??::${castAs}`, [key]) : key;
+
+    const columnKey = key.includes('->>') ? connection.raw(`??->>?`, key.split('->>')) : key;
+    const columnSelector = castAs ? connection.raw(`??::${castAs}`, [columnKey]) : columnKey;
```

Comment:
Still need to test to ensure that that `castAs` doesn't break when key includes `->>`